src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/cran/DescriptionParser.groovy

```diff
@@ -0,0 +1,130 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.blackducksoftware.integration.hub.detect.bomtool.cran;
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import com.blackducksoftware.integration.hub.bdio.simple.model.DependencyNode;
+import com.blackducksoftware.integration.hub.bdio.simple.model.Forge
+import com.blackducksoftware.integration.hub.detect.nameversion.NameVersionNode;
+import com.blackducksoftware.integration.hub.detect.nameversion.NameVersionNodeBuilder;
+import com.blackducksoftware.integration.hub.detect.nameversion.NameVersionNodeImpl;
+import com.blackducksoftware.integration.hub.detect.nameversion.NameVersionNodeTransformer
+
+public class DescriptionParser {
+    private final Logger logger = LoggerFactory.getLogger(PackRatNodeParser.class)
+
+    private NameVersionNode rootNameVersionNode
+    private NameVersionNodeBuilder nameVersionNodeBuilder
+    private HashSet<String> directDependencyNames
+    private NameVersionNode currentParent
+
+    private boolean inSpecsSection = false
+    private boolean inDependenciesSection = false
+
+    private NameVersionNodeTransformer nameVersionNodeTransformer
+    private DependencyNode rootProject
+    private final String packratLockContents
+    private Forge CRAN
+    private String lines
+
+
+    public DescriptionParser (){
+        //this.nameVersionNodeTransformer = nameVersionNodeTransformer
+        //this.CRAN = CRAN
+        // this.rootProject = rootProject
+        // this.packratLockContents = packratLockContents
+        //String[] lines = descriptionContents.split("\n")
+    }
+
+
+
+    String getProjectVersion(final String descriptionContents){
+        String[] lines = descriptionContents.split("\n")
+        String version;
+
+        for (String line : lines) {
+            //if (line != null) {
+            if (line.contains("Version")){
+                version = line.replace("Version: ", "").trim();
+            }
+        }
+        return version
+
+    }
+
+    void parseProjectDependencies(NameVersionNodeTransformer nameVersionNodeTransformer, DependencyNode rootProject, final String descriptionContents,  Forge CRAN) {
+        rootNameVersionNode = new NameVersionNodeImpl([name: rootProject.name, version: rootProject.version])
+        nameVersionNodeBuilder = new NameVersionNodeBuilder(rootNameVersionNode)
+        directDependencyNames = new HashSet<>()
+        currentParent = rootNameVersionNode
+
+        String[] lines = descriptionContents.split("\n")
+        String name;
+        String version;
+
+        for (String line : lines) {
+            //if (line != null) {
+            if (line.contains("Imports")) {
+                while (!line.contains("Suggestions") || line == "") {
+                    String[] imports = line.replace("Import","").trim().split(",")
+                }
```

Comment:
What is the point of this while loop? Why are you creating an array of strings and not doing anything with it?